samples/guide/src/test/kotlin/okhttp3/AllMainsTest.kt

```diff
@@ -0,0 +1,37 @@
+package okhttp3
+
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import java.io.File
+import java.lang.reflect.InvocationTargetException
+
+@RunWith(Parameterized::class)
+class AllMainsTest(val className: String) {
```

Comment:
This is really clever. I'd be tempted to just write 15 regular test methods, then you could assert exceptions here rather than in the samples themselves? These don't change too frequently!

Reply:
Left this mostly as is, my goal is keeping it stupidly simple to add a normal passing sample, but detect when that breaks or is flaky